addons/sale/tests/test_sale_order.py

```diff
@@ -73,6 +73,16 @@ def setUpClass(cls):
             'order_id': cls.sale_order.id,
             'tax_id': False,
         })
+        cls.sol_note = cls.env['sale.order.line'].create({
```

Comment:
Small addition to the validation review:

I would not create a note on the "main test invoice".
It changes the behavior of the "test_sale_order" where you need to increment each count by 1, when the purpose of that test is really to check that every line is correctly invoiced when it needs to be.

Instead, you could add a note line on the sale_order within your test method (test_so_note_to_invoice).
As changes are rollbacked between each test, it will not affect other test methods.

Reply:
👍 Implemented